tests/meta-mender-ci/recipes-testing/mender-coverage-client/mender-client_%.bbappend

```diff
@@ -0,0 +1,67 @@
+SUMMARY = "Build a Mender client which records coverage during its invocation"
+
+DEPENDS += "gobinarycoverage rsync-native"
+
+do_copy_src_to_build[nostamp] = "1"
+do_copy_src_to_build[doc] = "Copies the Mender client source from ${S}/src/${GO_IMPORT} \
+to ${B}/src/${GO_IMPORT}"
+do_copy_src_to_build[dirs] = "${B}src/github.com/mendersoftware/mender"
+do_copy_src_to_build () {
+    bbwarn "Copying source to the build dir..."
+    rsync -a ${S}/src/${GO_IMPORT}/ ${B}/src/${GO_IMPORT}/
+    cd ${B}/src/${GO_IMPORT}/
+    # Why is this needed?
+    git reset --hard HEAD
```

Comment:
I also don't see why! What happens if you remove it?

Reply:
Fails, with a dirty source dir when applying the patch